final_project/flaskr/schema.sql

```diff
@@ -2,50 +2,39 @@
 /* Why is 'status' blue 
 There are actually 3 states a bikerack can be in: "approved", "pending", and
 "rejected".
-for each vote also include it's time/date that it was voted
-separate table for votes, foreign key on bike rack id
-and this separate table has the data for voting: time/date, type of vote ("upvote" or "downvote"), index
-on the bikerack id (or time/date)
+separate table for votes, has foreign key on bike rack id
+and this separate table has the data for voting: type of vote (1 ("upvote") or -1 ("downvote")),
+user_id, index on the bikerack id (or user_id)
 The relationship between bikeracks and votes is one to many, one bikerack
 to many votes
 The bikeracks table is the parent table, and is the table that the
 foreign key constraint refers to.
 The votes table is the child table, which is the table that a foreign key
 consraint is applied.
 The rack_id in bikeracks is the parent key, the rack_id column in
-votes is called the child key.
-In sqlite the datatype of a value is associated with the value itself,
-not with its container */
-
-/* TODO Should I have an upvote count and downvote count in the bikeracks database?
-Then you don't have to count up the number of upvotes for a particular rack in the 
-votes table every time. */
+votes is called the child key.*/
 
 DROP TABLE IF EXISTS bikeracks;
 DROP TABLE IF EXISTS votes;
-DROP TABLE IF EXISTS users;
 DROP TABLE IF EXISTS bikeracks_history;
 DROP TABLE IF EXISTS votes_history;
-DROP TABLE IF EXISTS users_history; 
 
 CREATE TABLE bikeracks (
     rack_id INTEGER PRIMARY KEY AUTOINCREMENT,
     status TEXT DEFAULT "pending",
     latitude REAL NOT NULL,
     longitude REAL NOT NULL,
     address TEXT,
+    upvote_count INTEGER,
```

Comment:
These should be NOT NULL, no?
And DEFAULT 0 if that's not the default DEFAULT for integer columns.